git/git.go

```diff
@@ -1423,6 +1423,14 @@ func IsWorkingCopyDirty() (bool, error) {
 }
 
 func ObjectDatabase(osEnv, gitEnv Environment, gitdir, tempdir string) (*gitobj.ObjectDatabase, error) {
-	alternates, _ := cfg.OSEnv().Get("GIT_ALTERNATE_OBJECT_DIRECTORIES")
-	return gitobj.FromFilesystemWithAlternates(filepath.Join(dir, "objects"), cfg.TempDir(), alternates)
+	var options []gitobj.Option
+	alternates, _ := osEnv.Get("GIT_ALTERNATE_OBJECT_DIRECTORIES")
+	if alternates != "" {
+		options = append(options, gitobj.Alternates(alternates))
+	}
+	hashAlgo, _ := gitEnv.Get("extensions.objectformat")
+	if hashAlgo != "" {
+		options = append(options, gitobj.ObjectFormat(gitobj.ObjectFormatAlgorithm(hashAlgo)))
```

Comment:
IIUC, `hashAlgo` here is a string, which is cast to a `gitobj.ObjectFormatAlgorithm` type (which is a string [under the hood](https://github.com/git-lfs/gitobj/blob/add9e61146258df955958f6560361832257c8deb/object_db.go#L47)), which is passed to `gitobj.ObjectFormat()`, which returns a [setter function](https://github.com/git-lfs/gitobj/blob/add9e61146258df955958f6560361832257c8deb/object_db.go#L66-L68) that will be called in `gitobj.FromFilesystem()` to set its internal `args.objectFormat` to the given string value.  That in turn gets [passed](https://github.com/git-lfs/gitobj/blob/add9e61146258df955958f6560361832257c8deb/object_db.go#L82) to the `hasher()` function which [compares](https://github.com/git-lfs/gitobj/blob/add9e61146258df955958f6560361832257c8deb/object_db.go#L389-L396) the string to the two known hash names and returns an actual hash object.

What happens if the `hasher()` returns `nil`, though, because there's a misconfiguration and string pulled from the Git configs doesn't match either of the known hash functions?  Could that happen, and if so, what would be the effects?

Reply:
Good catch.  I'll add a check for this so we don't have a nil pointer dereference.